src/utils/mixins.js

```diff
@@ -245,23 +245,21 @@ const fileUploader = {
       this.errorMessage = null;
       const hookResult = await preUpload();
       this.indeterminate = false;
-      const promiseList = [];
-      // shallow copy into queue
-      const fileQueue = this.files.slice();
+      const results = [];
+      let i = 0;
       const WORKER_POOL_SIZE = 5;
       const workerPool = [...new Array(WORKER_POOL_SIZE)];
       await Promise.all(workerPool.map(async () => {
-        let file = fileQueue.shift();
-        while (file !== undefined) {
+        while (i < this.files.length) {
+          const file = this.files[i];
+          i += 1;
```

Comment:
```suggestion
          // eslint-disable-next-line no-plusplus
          const file = this.files[i++];
```
Unary is useful here.  I'd hate to have someone re-organize this function in such a way that breaks concurrency.  I don't know if I'd necessarily notice if someone accidentally moved Line 255 to the end of the loop, for example.

Reply:
Sure, I will add a code comment too, otherwise it is definitely a threat of someone thinking they can make this a for loop, or move the increment around.